__tests__/routes/courseRoutes.spec.js

```diff
@@ -319,157 +382,173 @@ describe('Course Routes', () => {
-        { _id: courseId },
-        { $push: { sections: sectionId } },
-        { returnDocument: 'after' } // 'after' returns the updated document
-      );
-      const updatedCourse = resultCourse.value;
+  it('should handle invalid course id when unsubscribing', async () => {
 
-      // Check if the section was successfully added
-      expect(updatedCourse.sections.find((element) => element == sectionId));
+    const user = await db.collection('users').findOne({ email: 'fake@gmail.com' });
+    const userId = user._id;
 
-      // send request where there are no sections in db that match the course's sections
-      const response = await request(`http://localhost:${PORT}`)
-        .get('/api/courses/' + courseId + '/sections');
+    // Simulate a request to unsubscribe with an invalid user id
+    const response = await request(`http://localhost:${PORT}`)
+      .post('/api/courses/this-is-aninvalid-courseId/unsubscribe')
+      .send({ user_id: userId });
 
-      expect(response.status).toBe(404);
-      expect(response.body.error.code).toBe('E0007');
-    });
+    expect(response.status).toBe(500);
+    expect(response.body.error.code).toBe('E0003');
   });
 
-
-  afterAll(async () => {
-    await db.collection('users').deleteMany({}); // Delete all documents in the 'users' collection
-    await db.collection('courses').deleteMany({}); // Delete all documents in the 'courses' collection
-    await db.collection('sections').deleteMany({}); // Delete all documents in the 'sections' collection
-    await server.close();
-    await mongoose.connection.close();
+  it('Decrements number of subscribers for a course, when unsubscribing', async () => {
+    // find a course and set number of subscription to 3
+    const courseUpdated = await db.collection('courses').findOneAndUpdate(
+      { title: 'test course' },
+      { $set: { numOfSubscriptions: 3 } },
+      { returnDocument: 'after' } // 'after' returns the updated document
+    );
+    const updatedCourse = courseUpdated.value;
+
+    // find a user and add the course to the user's subscriptions
+    const userUpdated = await db.collection('users').findOneAndUpdate(
+      { email: 'fake@gmail.com' },
+      { $push: { subscriptions: updatedCourse._id } },
+      { returnDocument: 'after' } // 'after' returns the updated document
+    );
+    const updatedUser = userUpdated.value;
+
+    const response = await request(`http://localhost:${PORT}`)
+      .post('/api/courses/' + updatedCourse._id + '/unsubscribe')
+      .send({ user_id: updatedUser._id });
+    // check the post request is successful
+    expect(response.status).toBe(200);
+
+    // check the number of subscribers is decremented
+    const courseNew = await db.collection('courses').findOne({ _id: updatedCourse._id });
+    const userNew = await db.collection('users').findOne({ _id: updatedUser._id });
+    expect(courseNew.numOfSubscriptions).toBe(2);
+    expect(userNew.subscriptions).toHaveLength(0);
   });
```

Comment:
File should be renamed to courseRoutes.spec.js

Reply:
I know but there are two group that have made test for courses and one is already .test, i tried to merge them into one file but got errors... I am not sure if we should name them something else or figure out a way of merging them

Reply:
@sofiiagran & @JackSMunn I have now merged the test file into a courseRoutes.test.js (hopefully this will be the last time this has to be consolidated 🥲